ArduSub/AP_Arming_Sub.cpp

```diff
@@ -17,6 +17,40 @@ bool AP_Arming_Sub::pre_arm_checks(bool display_failure)
     if (armed) {
         return true;
     }
+    
+    // if there's no shift assigned, it must not be a shift funtion
+    bool has_shift_function = false;
+    bool has_disarm_function = false;
+    // make sure the craft has a disarm button assigned before it is armed
+    // check all the standard btn functions
+    for (int i = 0; i < 16; i++) {
+        int btn = sub.get_button(i)->function(false);
+
+        if (btn == JSButton::k_arm_toggle || btn == JSButton::k_disarm) {
+            has_disarm_function = true;
+            break;
+        }
+        if (btn == JSButton::k_shift) {
+            has_shift_function = true;
+        }
+    }
+    // check all the shift functions if there's shift assigned
+    if (has_shift_function && !has_disarm_function) {
+        for (int i = 0; i < 16; i++) {
+            int btn = sub.get_button(i)->function(true);
+
+            if (btn == JSButton::k_arm_toggle || btn == JSButton::k_disarm) {
+                has_disarm_function = true;
+                break;
+            }
+        }
+    }
+
+    // don't allow arming unless there is a disarm button configured
+    if (!has_disarm_function) {
```

Comment:
This should be a function call - break out a `has_disarm_function()`, which does a return in place of the `break`s in those loops.